### PR TITLE
List module functions documentation

### PR DESCRIPTION
**Pull request by onpikono**
I have updated initial bunch of `List` functions with the goal to:

- unify the language and layout (subjective)
- ensure all `<code>` blocks have `lang="fsharp"` attribute
- fix any compilation errors
- correct code to follow F# style guide
- correct spelling mistakes

I would appreciate feedback to any (type of) change as some of them may be subjective and the community may view them differently.

Once we have basics agreed, I will proceed further.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/fsharp?pullRequest=12319) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/fsharp?pullRequest=12319) before we can accept your contribution.<br/><br/>:x: onpikono [sign now](https://cla.dotnetfoundation.org/dotnet/fsharp?pullRequest=12319)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/fsharp?pullRequest=12319) it.</sub>

File: src/fsharp/FSharp.Core/list.fsi [hunk 1]
```diff
@@ -124,69 +117,68 @@ module List =
...
-    /// <example id="choose-2">
-    /// Using the identity function <c>id</c> (is defined like <c>fun x -> x</c>):
-    /// <code lang="fsharp">
-    /// 
-    /// let input1 = [ Some 1; None; Some 3; None ]
-    ///
-    /// input1 |> List.choose id  // evaluates [1; 3]
-    /// </code>
-    /// </example>
-    /// <example id="choose-3">
```
Comment (dsyme) [inline src/fsharp/FSharp.Core/list.fsi:172]: Why did you remove two samples here?

**Comment by dsyme** [comment]
On the whole this looks fine. A couple of samples were removed, I'm not sure why?  

On the whole I was less sure that the changes to the "<summary>" fields were good - all the rest of the changes looked fine

**Review by dsyme — COMMENTED** [review]
see comments

**Comment by dsyme** [comment]
@onpikono Can you reply to my comments here please?  Thanks :)

File: src/fsharp/FSharp.Core/list.fsi (hunk 1 above)
Comment (KevinRansom) [reply src/fsharp/FSharp.Core/list.fsi:172]: @dsyme I added them back
Kevin

**Review by dsyme — APPROVED** [review]

**Comment by KevinRansom** [comment]
@onpikono 

Thanks for taking care of this

Kevin
